syntax = "proto3";

package HelloWorld;

service Greeter
{
    rpc SayHello(HelloRequest) returns(HelloReply) {}
    rpc SayHelloServerStream(HelloRequest) returns(stream HelloReply) {}
    rpc SayHelloClientStream(stream HelloRequest) returns(HelloReply) {}
    rpc SayHelloBidiStream(stream HelloRequest) returns(stream HelloReply) {}
}

message HelloRequest
{
    string name = 1;
}

message HelloReply
{
    string message = 1;
}
